jax/_src/lax/slicing.py

```diff
@@ -684,7 +689,7 @@ def index_in_dim(operand: Array, index: int, axis: int = 0,
 def dynamic_slice_in_dim(operand: Array, start_index: Array,
                          slice_size: int, axis: int = 0) -> Array:
   """Convenience wrapper around dynamic_slice applying to one dimension."""
-  start_indices = [lax._zero(start_index)] * operand.ndim
+  start_indices = [0] * operand.ndim
```

Comment:
I understand why we are doing this now, but why didn't we do it before?

Reply:
Indeed... 🤷‍♂️ 

Hope past versions of ourselves weren't being too clever.

Reply:
Revising...